tests/README.md

```diff
@@ -0,0 +1,104 @@
+# 🧪 Testing cheatsheet
+
+## :scroll: Create a test file
+### C++
+1. `FIMS:::use_gtest_template()`: Create a new file `tests/gtest/test_class_function.cpp` 
+for GoogleTest and register the test in the `tests/gtest/CMakeLists.txt` file.
+
+### R 
+1. `FIMS:::use_testthat_template()`: Create a new file `tests/testthat/test-function_name.R` 
+for {testthat}.
+
+## 🛢️ Prepare test data
+### In-line test data
+- Write setup code directly in the `setup` section of the test file.
+
+### C++
+1. **Reusable test data**: Create a `tests/gtest/test_XXX_test_fixture.hpp` 
+file and write code to prepare test data.
+1. **Integration test data**: Create a `tests/gtest/integration/XXX.hpp` 
+file and write code to prepare test data.
+
+### R reusable test data
+1. Write code to `prepare_test_data()` in 
+`tests/testthat/helper-integration-tests-setup-run.R`.
+1. If package data are needed, make sure 
+`tests/testthat/fixtures/integration_test_data_components.RData` and 
+`tests/testthat/fixtures/integration_test_data.RData` are preloaded in 
+`prepare_test_data()` as both are generated by running `R/data1.R`.
+1. Save test data with 
+`base::saveRDS(object, file = testthat::test_path("fixtures", "data_name.RDS"))`.
+1. In test file that requires the test data, call `prepare_test_data()` to load 
+the data:
+```r
+if (!file.exists(test_path("fixtures", "data_name.RDS"))) {
+  prepare_test_data()
+}
+```
+1. `FIMS:::remove_test_data()`: Remove test data from `tests/testthat/fixtures` 
+to run the tests from fresh if changes in the core code affect FIMS input or 
+output.
+
+## :hammer: Helper functions to set up tests
+### R integration tests
+1. `setup_and_run_FIMS_without_wrappers()`: Set up a FIMS model without wrappers 
+(in `tests/testthat/helper-integration-tests-setup-run.R`).
+1. `setup_and_run_FIMS_with_wrappers()`: Set up a FIMS model with wrappers (in 
+`tests/testthat/helper-integration-tests-setup-run.R`).
+1. `verify_fims_deterministic()`: Compare the FIMS model deterministic run output
+against the expected "truth" from an operating model (in
+`tests/testthat/helper-integration-tests-validation.R`).
+1. `verify_fims_nll()`: Compare the FIMS model NLL against the expected NLL
+calculated from an operating model (in
+`tests/testthat/helper-integration-tests-validation.R`).
+1. `validate_fims()`: Compare the FIMS model estimation run output against the 
+expected "truth" from an operating model (in
+`tests/testthat/helper-integration-tests-validation.R`)
+
+## :computer: Run tests
+### C++
+1. `FIMS:::setup_and_run_gtest()`: Set up integration test data and run the 
+GoogleTest suite.
+1. `FIMS:::run_gtest()`: Run the GoogleTest suite.
+
+### R
+1. Add `testthat::skip_on_ci()` at the beginning of a test file to skip it during
+continuous integration runs.
+1. Add `testthat::skip_on_covr()` at the beginning of a test file to skip it
+during coverage calculation.
+1. Run `devtools::test()` to execute all tests in the package.
```

Comment:
- Update line 69: run `devtools::load()` before running `devtools::test()`. 
- Add a bullet point to mention running `devtools::install()` before running `test-parallel-*.R`